RFC Draft.md

```diff
@@ -0,0 +1,583 @@
+
+```Typescript
+
+const resp = await confirmSignUp({
+  username: "username",
+  confirmationCode: "112233",
+});
+
+if (resp.isSignUpComplete) {
+  // Show login component
+}
+
+```
+
+**Proposed Amplify v6 DX**
+
+<img width="594" alt="confirmSignUp-v6" src="https://user-images.githubusercontent.com/70438514/222456623-34d48988-5e3d-43be-bc9a-7c6d8ecb0b00.png">
+
+Try out the new types here: https://stackblitz.com/edit/rfc-auth?file=auth.ts
+
+# `Storage` Category Changes
+
+Amplify is proposing the following changes for the `Storage` category.
+
+## Introduction of object reference types
+
+In order to permit better inoperability between `storage` APIs and to enable future improvements such as more granular bucket management, we will introduce `StorageObjectReference` & `StoragePrefixReference` types to represent items in cloud storage. An example for copying an object from one access level to another is highlighted below.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```TypeScript
+// List all public photos
+const listResponse = await Storage.list('photos/', { level: 'public' });
+const firstPhoto = listResponse.results?.[0];
+
+// Copy the first photo returned to the current user's private prefix
+if (firstPhoto) {
+  await Storage.copy({
+    {
+      key: firstPhoto.key,
+      level: 'public'
+    },
+    {
+      key: firstPhoto.key,
+      level: 'private'
+    }
+  })
+}
+```
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```TypeScript
+// List all public photos
+const listResponse = await Storage.list({
+  key: getPrefixReference('photos/', { level: 'public' })
+})
+const firstPhoto = listResponse.files?.[0];
+
+/*
+As a note, APIs will allow customers to specify keys by string if they do not need to override the access level. For
+example, the following operation will list all files for the current user.
+*/
+const listResponseDefault = await Storage.list({
+  key: 'photos/'
+})
+
+// Copy the first photo returned to the current user's private prefix
+if (firstPhoto) {
+  await copy({
+    source: firstPhoto,
+    destination: copyObjectReference(firstPhoto, { level: 'private' }),
+  });
+}
+```
+
+## Splitting up the `get` API
+
+To better capture customer intent and simplify API types we will split up the `get` API into `getUrl` & `download`. An example for generating a pre-signed URL & downloading a file from the results of a `list` operation is highlighted below.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```TypeScript
+// List public photos
+const listResponse = await Storage.list('photos/', { level: 'public' });
+const firstPhoto = listResponse.results?.[0];
+
+// Generate a pre-signed URL for a file
+const presignedUrl = await Storage.get(firstPhoto.key, { level: 'public' });
+
+// Download a file
+const downloadResult = await Storage.get(firstPhoto.key, { download: true, level: 'public' });
+
+// Customer provided utility for handling downloaded file
+downloadBlob(downloadResult.Body, 'download.jpg');
```

Comment:
I see this in our examples, but I'm not sure contextually what its telling me here.